src/main/java/edu/ucsb/cs156/organic/controllers/StaffController.java

```
@@ -0,0 +1,116 @@
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+import edu.ucsb.cs156.organic.entities.Staff;
+import edu.ucsb.cs156.organic.entities.User;
+import edu.ucsb.cs156.organic.errors.EntityNotFoundException;
+import edu.ucsb.cs156.organic.repositories.StaffRepository;
+import edu.ucsb.cs156.organic.repositories.UserRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+
+@Tag(name = "Staff")
+@RequestMapping("/api/staff")
+@RestController
+@Slf4j
+public class StaffController extends ApiController{
+
+    @Autowired
+    StaffRepository staffRepository;
+
+    @Autowired
+    UserRepository userRepository;
+
+    @Operation(summary = "List all staff")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @GetMapping("/all")
+    public Iterable<Staff> allCourses() {
+        User u = getCurrentUser().getUser();
+        log.info("u={}", u);
+
+        return staffRepository.findAll();
+    }
+
+    @Operation(summary = "Create a new staff")
+    @PreAuthorize("hasAnyRole('ROLE_ADMIN')")
+    @PostMapping("")
+    public Staff postCourse(
+            @Parameter(name = "courseId", description = "course id") @RequestParam Long courseId,
+            @Parameter(name = "githubId", description = "school abbreviation e.g. UCSB") @RequestParam Integer githubId)
+            throws JsonProcessingException {
+
+
```

You need to add a check here that checks whether the githubId exists in the Users table.  If not, you need to return an error.  See, for example, this code in the `CoursesController`.  This uses the `Course` table to look up a course, but it's the same idea; you'll just be looking up a `User` by `githubId`:

```
   Course course = courseRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException(Course.class, id.toString()));
```

---

And for that matter, you also need to be validating the course id, which you can do with *exactly* this code.

---

Fixed it for both post and update